libs/permissions-client/test/permissions_client/integration_test.clj

```diff
@@ -0,0 +1,267 @@
+(ns permissions-client.integration-test
+  (:require [permissions-client.core :as pc])
+  (:use [clojure.test]))
+
+(def ^:dynamic base-uri "http://permissions:60000/")
+
+(defn create-permissions-client []
+  (pc/new-permissions-client base-uri))
+
+(defn run-integration-tests [f]
+  (when (System/getenv "RUN_INTEGRATION_TESTS")
+    (f)))
+
+(defn with-base-uri [f]
+  (if-let [uri (System/getenv "PERMISSIONS_BASE_URI")]
+    (binding [base-uri uri]
+      (f))
+    (f)))
+
+(defn add-test-subjects []
+  (let [client (create-permissions-client)]
+    (pc/add-subject client "ipcdev" "user")
+    (pc/add-subject client "ipctest" "user")
+    (pc/add-subject client "ipcusers" "group")))
+
+(defn remove-test-subjects []
+  (let [client (create-permissions-client)]
+    (dorun (map (comp (partial pc/delete-subject client) :id)
+                (:subjects (pc/list-subjects client))))))
+
+(defn add-test-resources []
+  (let [client (create-permissions-client)]
+    (pc/add-resource client "a" "app")
+    (pc/add-resource client "b" "app")
+    (pc/add-resource client "C" "analysis")
+    (pc/add-resource client "D" "analysis")))
+
+(defn remove-test-resources []
+  (let [client (create-permissions-client)]
+    (dorun (map (comp (partial pc/delete-resource client) :id)
+                (:resources (pc/list-resources client))))))
+
+(defn remove-test-resource-types []
+  (let [client (create-permissions-client)]
+    (dorun (map (comp (partial pc/delete-resource-type client) :id)
+                (remove (comp #{"app" "analysis"} :name)
+                        (:resource_types (pc/list-resource-types client)))))))
+
+(defn add-test-permissions []
+  (let [client (create-permissions-client)]
+    (pc/grant-permission client "app" "a" "user" "ipcdev" "own")
+    (pc/grant-permission client "app" "b" "user" "ipctest" "write")
+    (pc/grant-permission client "app" "a" "group" "ipcusers" "read")
+    (pc/grant-permission client "app" "b" "group" "ipcusers" "admin")
+    (pc/grant-permission client "analysis" "C" "user" "ipcdev" "own")
+    (pc/grant-permission client "analysis" "D" "user" "ipctest" "own")
+    (pc/grant-permission client "analysis" "C" "group" "ipcusers" "read")))
+
+(defn permission-index
+  [perm]
+  (mapv (partial get-in perm)
+        [[:resource :resource_type] [:resource :name] [:subject :subject_type] [:subject :subject_id]]))
+
+(defn remove-test-permissions []
+  (let [client (create-permissions-client)]
+    (dorun (map (comp (partial apply pc/revoke-permission client) permission-index)
+                (:permissions (pc/list-permissions client))))))
+
+(defn with-test-data [f]
+  (add-test-subjects)
+  (add-test-resources)
+  (add-test-permissions)
+  (f)
+  (remove-test-subjects)
+  (remove-test-resources)
+  (remove-test-resource-types)
+  (remove-test-permissions))
+
+(use-fixtures :each run-integration-tests with-base-uri with-test-data)
```

Comment:
Nice! 👍
I like that we can run integration tests like this with clojure, rather than depending on external scripts.